src/interpreter/mod.rs

```diff
@@ -100,6 +99,39 @@ impl<'txin> Interpreter<'txin> {
             has_errored: false,
         }
     }
+
+    /// Outputs a "descriptor" string which reproduces the spent coins
+    ///
+    /// This may not represent the original descriptor used to produce the transaction,
+    /// since it cannot distinguish between sorted and unsorted multisigs (and anyway
+    /// it can only see the final keys, keyorigin info is lost in serializing to Bitcoin).
+    ///
+    /// If you are using the interpreter as a sanity check on a transaction,
+    /// it is worthwhile to try to parse this as a descriptor using `from_str`
+    /// which will check standardness and consensus limits, which the interpreter
+    /// does not do on its own. Or use the `inferred_descriptor` method which
+    /// does this for you.
+    pub fn inferred_descriptor_string(&self) -> String {
+        match self.inner {
+            inner::Inner::PublicKey(ref pk, inner::PubkeyType::Pk) => format!("pk({})", pk),
+            inner::Inner::PublicKey(ref pk, inner::PubkeyType::Pkh) => format!("pkh({})", pk),
+            inner::Inner::PublicKey(ref pk, inner::PubkeyType::Wpkh) => format!("wpkh({})", pk),
```

Comment:
The public key one is lacking ShWpkh. I think we are not parsing it in the `from_txdata` function probably a few commits back.

Reply:
Oh, yeah, I totally forgot that was a thing...yeah, we were not parsing it in the original code.